hollow/src/main/java/com/netflix/hollow/api/client/HollowDataHolder.java

```diff
@@ -124,15 +124,16 @@ private void applySnapshotTransition(HollowConsumer.Blob snapshotBlob, HollowCon
     }
 
     private void initializeAPI() {
-        if(objLongevityConfig.enableLongLivedObjectSupport()) {
+        if(objLongevityConfig != null && objLongevityConfig.enableLongLivedObjectSupport()) {
```

Comment:
Consider adding a private method `isObjectLongevityEnabled` rather than duplicating the logic.

I presume `staleReferenceDetector` is intertwined with `objLongevityConfig`. Is it also connected to the value returned by `objLongevityConfig.enableLongLivedObjectSupport`?